docs/argocd.md

```diff
@@ -0,0 +1,207 @@
+# Integrating GitOps Flow with the SiteConfig Operator
+
+This guide provides instructions for integrating a GitOps workflow with the SiteConfig operator.
+We recommend deploying the ArgoCD operator using the
+[Red Hat OpenShift GitOps operator](https://catalog.redhat.com/software/operators/detail/5fb288c70a12d20cbecc6056).
+
+## Table of Contents
+1. [Prepare Git Repository](#prepare-git-repository)
+2. [Configure ArgoCD](#configuring-argocd)
+    - [Create ArgoCD Project](#create-argocd-project)
+    - [Grant ArgoCD Permissions](#grant-argocd-permissions)
+    - [Configure Access to the Git Repository](#configure-access-to-the-git-repository)
+    - [Create ArgoCD Application](#create-argocd-application)
+
+## Prepare Git Repository
+
+To start, create a Git repository with a directory structure similar to the suggested example shown below.
+
+```sh
+clusters/
+ |- spoke1/
+ |-- extra-manifests-cm1.yaml
+ |-- extra-manifests-cm2.yaml
```

Comment:
Typically the extra manifests CM will be shared across many spoke clusters. Can you adjust these to reflect that?

Reply:
The extra manifests ConfigMaps need to be created in the same namespace as the `ClusterInstance` CR (i.e. local reference).

Reply:
Doc is ok for now but this is a limitation that we will need to address in the dev/tech preview phases. The set of extra manifests will be consistent across many clusters (1000's) and we will need the ability to keep a central referenced version.
Can you please open a story to track this?

Reply:
@imiller0 I had created a [PR](https://github.com/stolostron/siteconfig/pull/7) to address this earlier on.